### ISSUE TITLE
New bigquery Union modifiers

### ISSUE THREAD

**Issue by HampB** [sqlfluff/sqlfluff#6643]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Here's a more structured and detailed version of your feature request:  

---

#### Background  
BigQuery has introduced new modifiers for `UNION` statements, as documented [here](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#union). These modifiers, such as `INNER UNION BY`, allow more precise control over how rows are combined.  

#### Issue  
Currently, SQLFluff does not support these new `UNION` syntax extensions, resulting in parsing errors for valid BigQuery SQL statements like:  

```sql
SELECT * FROM table1
INNER UNION ALL BY name
SELECT * FROM table2;
```  

#### Expected Behavior  
SQLFluff should correctly parse and lint BigQuery queries that use `INNER UNION BY` and similar modifiers without errors.  

---
If it's a simple fix - i could probably contribute with some guidance

### Dialect

Bigquery

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by semihtekten** [comment]
agreed. can't use this new feature.
